### ISSUE TITLE
Option to remove KeyInfo (or replace X509Data with KeyName) from signature

### ISSUE THREAD

**Issue by rcwinder** [spring-projects/spring-security#12179]
The SAML IDP that we use requires us to follow these rules:
When signing metadata, it should contain the <Signature> element, and it MUST contain only an <X509Data> element with an <X509Certificate> element.
Signing for all other cases (Authn, artifact resolve, logout, etc): The <Signature> element MAY <KeyInfo> element that contains a <KeyName>. The signature MUST NOT contain other elements (such as <X509Data>). If a <KeyInfo> element is not included in the message, the metadata should contain a valid certificate to validate the message.

In short, currently spring security always provides a KeyInfo element when signing occurs, and always does this with the x509 data + certificate. In our case, it would be required to change this, and either remove the KeyInfo element entirely, or replace the X509Data with a KeyName. 
Hopefully this can be made configurable for the RelyingPartyRegistration. 

PS. I found this issue, this is possibly why we didnt run into this issue sooner: 
(https://github.com/spring-projects/spring-security/issues/11354)

**Comment by jzheaux** [comment]
@rcwinder, thanks for the report. It looks like something may have caused some of the words to drop in your description and I can't completely follow your situation. Here is an example of something in your description where it appears there is missing detail.

> The signature MUST NOT contain other elements (such as )

Will you please go back through your description and ensure that it is complete? After that, I think I'll be able to help better.

**Comment by spring-projects-issues** [comment]
If you would like us to look at this issue, please provide the requested information. If the information is not provided within the next 7 days this issue will be closed.

**Comment by spring-projects-issues** [comment]
Closing due to lack of requested feedback. If you would like us to look at this issue, please provide the requested information and we will re-open the issue.
